src/codegeneration/ClassTransformer.js

```diff
@@ -448,34 +447,19 @@ export class ClassTransformer extends TempVarTransformer {
                                   constructorParams, null, [], constructorBody);
   }
 
-  /**
-   *  Appends assignment to this for the member variables after the super call.
-   */
   appendInstanceInitializers_(constructor, initStatements, superClass) {
     if (initStatements.length === 0) return constructor;
 
-    let statements = constructor.body.statements;
-
     if (superClass) {
-      let superExpressionIndex = -1;
-
-      for (let i = 0; i < statements.length; i++) {
-        let statement = statements[i];
-        if (statement.isDirectivePrologue()) {
-          continue;
-        }
-        if (isSuper(statement)) {
-          superExpressionIndex = i;
-          break;
-        }
-      }
-
-      statements = statements.slice();
-      statements.splice(superExpressionIndex + 1, 0, ...initStatements);
-    } else {
-      statements = prependStatements(statements, ...initStatements);
+      let initExpression = createCommaExpression(
+          initStatements.map((statement) => statement.expression));
```

Comment:
that's probably something we could do - currently we use initStatements below (l 461) and for getDefaultCtor